src/libraries/Common/src/System/Net/StreamBuffer.cs

```diff
@@ -292,7 +292,10 @@ public void Dispose()
             AbortRead();
             EndWrite();
 
-            _buffer.Dispose();
+            lock (SyncObject)
```

Comment:
Was the lack of this previously a bug, or is there something about switching to use MultiArrayBuffer that requires this?

Reply:
It was previously a bug, but the window was tiny. With MultiArrayBuffer, the window is a bit bigger.